miden-lib/asm/miden/kernels/tx/account.masm

```diff
@@ -344,22 +354,29 @@ end
 #! - V' is the value to set.
 #! - V is the previous value of the item.
 export.set_item
-    # TODO: we execute `push.1 drop` before `emit` as decorators are not supported without other
-    #       instructions - see: https://github.com/0xPolygonMiden/miden-vm/issues/1122
-    # emit event to signal that an account storage item is being updated
-    push.1 drop emit.ACCOUNT_STORAGE_SET_ITEM_EVENT
+    # TODO: we execute `push.0 drop` before `emit` as decorators are not supported without other
+    # instructions - see: https://github.com/0xPolygonMiden/miden-vm/issues/1122
+    push.0 drop emit.ACCOUNT_STORAGE_BEFORE_SET_ITEM_EVENT 
+
+    # duplicate the index and the V' to be able to emit an event after an account storage item is 
+    # being updated
+    movdn.4 dupw dup.8
+    # => [index, V', V', index]
 
     # get the storage root
     exec.memory::get_acct_storage_root
-    # => [R, index, V']
+    # => [R, index, V', V', index]
 
     # set the item in storage
     movup.4 push.STORAGE_TREE_DEPTH mtree_set
-    # => [V, R']
+    # => [V, R', V', index]
 
     # set the new storage root
     swapw exec.memory::set_acct_storage_root
-    # => [V]
+    # => [V, V', index]
+
+    # emit event to signal that an account storage item is being updated
+    swapw movup.8 emit.ACCOUNT_STORAGE_AFTER_SET_ITEM_EVENT drop dropw
```

Comment:
nit: let's add a comment with the stack state after this line, which I think would be just:
```
# => [V]
```

Reply:
By the way, sometimes we write stack state with ellipsis at the end (e.g. `# => [KEY, NEW_VALUE, index, ...]`), but sometimes without it. Should we stick to the same standard in all places? I think we can omit this ellipsis since it always means that there are zeros ahead.